protocol/v2/ssv/validator/non_committee_validator.go

```diff
@@ -33,69 +58,93 @@ func NewNonCommitteeValidator(logger *zap.Logger, identifier spectypes.MessageID
-		signedMsg := &spectypes.SignedSSVMessage{}
-		if err := signedMsg.Decode(msg.SSVMessage.GetData()); err != nil {
-			logger.Debug("❗ failed to get consensus Message from network Message", zap.Error(err))
-			return
-		}
+	if len(quorums) == 0 {
+		return
+	}
 
-		decMsg, err := specqbft.DecodeMessage(signedMsg.SSVMessage.Data)
-		if err != nil {
-			logger.Debug("❗ failed to decode message", zap.Error(err))
+	for key, quorum := range quorums {
+		role := ncv.getRole(msg, key.Root)
+		println("<<<<<<<<<<<<<<<<<<<<<<<<works>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>")
+		println(role.String())
+		println("<<<<<<<<<<<<<<<<<<<<<<<<works>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>>")
+		validator := ncv.ValidatorStore.ValidatorByIndex(key.ValidatorIndex)
+		MsgID := exporter_message.NewMsgID(ncv.qbftController.GetConfig().GetSignatureDomainType(), validator.ValidatorPubKey[:], role)
+		if err := ncv.Storage.Get(MsgID.GetRoleType()).SaveParticipants(MsgID, slot, quorum); err != nil {
+			logger.Error("❌ could not save participants", zap.Error(err))
 			return
+		} else {
+			var operatorIDs []string
+			for _, share := range quorum {
+				operatorIDs = append(operatorIDs, strconv.FormatUint(share, 10))
+			}
+			logger.Info("✅ saved participants",
+				zap.String("converted_role", role.ToBeaconRole()),
+				zap.String("validator_index", strconv.FormatUint(uint64(key.ValidatorIndex), 10)),
+				zap.String("signers", strings.Join(operatorIDs, ", ")),
+			)
 		}
 
-		// only supports decided msg's
-		if decMsg.MsgType != specqbft.CommitMsgType || !ncv.CommitteeMember.HasQuorum(len(signedMsg.OperatorIDs)) {
-			return
+		if ncv.newDecidedHandler != nil {
+			ncv.newDecidedHandler(qbftstorage.ParticipantsRangeEntry{
+				Slot:       slot,
+				Signers:    quorum,
+				Identifier: MsgID,
+			})
 		}
+	}
+}
 
-		logger = logger.With(fields.Height(decMsg.Height))
-
-		if decided, err := ncv.qbftController.ProcessMsg(logger, signedMsg); err != nil {
-			logger.Debug("❌ failed to process message",
-				zap.Uint64("msg_height", uint64(decMsg.Height)),
-				zap.Any("signers", signedMsg.OperatorIDs),
-				zap.Error(err))
-		} else if decided != nil {
-			if inst := ncv.qbftController.StoredInstances.FindInstance(decMsg.Height); inst != nil {
-				logger := logger.With(
-					zap.Uint64("msg_height", uint64(decMsg.Height)),
-					zap.Uint64("ctrl_height", uint64(ncv.qbftController.Height)),
-					zap.Any("signers", signedMsg.OperatorIDs),
-				)
-				if err = ncv.qbftController.SaveInstance(inst, signedMsg); err != nil {
-					logger.Debug("❗failed to save instance", zap.Error(err))
-				} else {
-					logger.Debug("💾 saved instance")
-				}
-			}
+func (ncv *CommitteeObserver) getRole(msg *queue.DecodedSSVMessage, root [32]byte) exporter_message.RunnerRole {
+	if msg.MsgID.GetRoleType() == spectypes.RoleCommittee {
+		_, found := ncv.Roots[root]
+		if !found {
+			return exporter_message.RoleAttester
 		}
-		return
+		return exporter_message.RoleSyncCommittee
```

Comment:
this looks a bit hacky. so you save only sync commitee roots? but what if we have not root at all for this, we'll say its attester..

Reply:
@y0sher didn't understand